_extensions/d2/d2.lua

```diff
@@ -121,13 +129,17 @@ local function render_graph(globalOptions)
 
       -- Set default filename
       if options.filename == nil then
-        options.filename = "diagram-" .. counter
+        options.filename = "diagram-" .. counter .. math.random (1, 1000)
```

Comment:
Why add a random number to this filename?

Reply:
I had an issue with the extension *not* overwriting existing files that this seemed to resolve. I’ll try to put together a reprex (it may be specifically related to using knit_child?) to see if this random number assignment is a real fix or if there may be a better way to fix the issue.